tests/tests.rs

```diff
@@ -154,9 +155,44 @@ fn run_compile_test(
         }
     };
 
-    output.push(format!("{}.{}", name, ext));
+    let header_suffix = ".header";
+    let header_position = name.rfind(&header_suffix);
+    let is_header = header_position.is_some();
+
+    let source_file = format!(
+        "{}.{}",
+        if is_header {
+            &name[0..header_position.unwrap()]
+        } else {
+            name
+        },
+        &ext
+    );
+    output.push(&source_file);
+
+    if is_header {
+        let header_ext = match output.extension().unwrap().to_str().unwrap().as_ref() {
+            "cpp" => "hpp",
+            "c" => "h",
+            _ => "",
```

Comment:
May be better to panic here or something, or just default to `.h`.

Reply:
ok